mindsdb/api/mysql/mysql_proxy/utilities/sql.py

```diff
@@ -72,11 +70,10 @@ def adapt_query(node, is_table, **kwargs):
         )
         query_str = render.get_string(query_ast, with_failback=True)
 
-    # workaround to prevent duckdb.TypeMismatchException: serialize and deserialize with feather
+    # workaround to prevent duckdb.TypeMismatchException
     if len(df) > 0 and table_name.lower() in ('models', 'predictors'):
-        fd = io.BytesIO()
-        df.to_feather(fd)
-        df = pd.read_feather(fd)
+        if 'TRAINING_OPTIONS' in df.columns:
```

Comment:
This effectively fixes the issue with nested dicts, but how about the three lines you've removed here? Shouldn't they stay? If not, then feel free to merge 👍

Reply:
removed lines was cause of error:
`pyarrow.lib.ArrowInvalid: ('cannot mix list and non-list, non-null values', 'Conversion failed for column TRAINING_OPTIONS with type object')`